strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/main/java/org/carlspring/strongbox/artifact/coordinates/NpmArtifactCoordinates.java

```diff
@@ -158,4 +169,13 @@ public static NpmArtifactCoordinates of(String scope,
         return new NpmArtifactCoordinates(scope, name, version);
     }
 
+    public static NpmArtifactCoordinates of(URI resource)
+    {
+        return parse(resource.toString());
+    }
+    
+    public static void main(String args[]) throws Exception{
```

Comment:
Could you please extract this to a proper test, if it's still required?

Reply:
done